app/controllers/application_controller.rb

```diff
@@ -1,10 +1,14 @@
 class ApplicationController < ActionController::Base
   protect_from_forgery with: :exception
 
-  helper_method :cart
+  helper_method :cart, :current_user
 
   def cart
     @cart ||= Cart.new(session[:cart] ||= Hash.new(0))
   end
 
+  def current_user
+    @user ||= User.find(session[:user_id]) if session[:user_id]
+  end
+
```

Comment:
Looks great. I was thinking along these lines before we left today but definitely interested to hear how you thought through it